packages/ai-jsx/src/batteries/use-tools.tsx

```diff
@@ -203,30 +211,31 @@ export async function* UseTools(props: UseToolsProps, { render }: RenderContext)
 }
 
 /** @hidden */
-export async function* UseToolsFunctionCall(props: UseToolsProps, { render }: RenderContext) {
-  const messages = [
-    <SystemMessage>You are a smart agent that may use functions to answer a user question.</SystemMessage>,
-  ];
-  if (props.fallback) {
-    messages.push(
-      <SystemMessage>Here's the fallback strategy/message if something failed: {props.fallback}</SystemMessage>
-    );
-  }
-  messages.push(<UserMessage>{props.query}</UserMessage>);
+export async function* UseToolsFunctionCall(props: UseToolsProps, { render }: RenderContext): RenderableStream {
+  yield AppendOnlyStream;
 
-  do {
-    const modelResponse = <ChatCompletion functionDefinitions={props.tools}>{messages}</ChatCompletion>;
+  const conversation = [memo(props.children)];
```

Comment:
Oh nice, I like that you are memoizing it. Never thought of that. This won't render the children, right?